### PR TITLE
chore(refactor): Make separate blueprint for events and orders

### LINKED ISSUE #6607: Separate order and event blueprints
**Describe the bug**
Right now even the custom endpoints of orders and events have been extended from ticket_blueprint.

**Additional context**
On it.

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #6607 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/custom/events.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        event_id = event_invoice.event_id
+    except NoResultFound:
+        return NotFoundError({'source': ''}, 'Event Invoice not found').respond()
+    if not current_user.is_organizer(event_id) and not current_user.is_staff:
+        return ForbiddenError({'source': ''}, 'Unauthorized Access').respond()
+    key = UPLOAD_PATHS['pdf']['event_invoices'].format(identifier=invoice_identifier)
+    file_path = '../generated/invoices/{}/{}/'.format(key, generate_hash(key)) + invoice_identifier + '.pdf'
+    try:
+        return return_file('event-invoice', file_path, invoice_identifier)
+    except FileNotFoundError:
```
Comment (niranjan94) [inline app/api/custom/events.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4554924)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 3
           

Complexity increasing per file
==============================
- app/api/custom/tickets.py  5
- app/api/custom/events.py  6
- app/api/custom/orders.py  5
         

Clones added
============
- app/api/custom/orders.py  2
         

Clones removed
==============
+ app/api/auth.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-server/open-event-orga-server/pullRequest?prid=4554924&bid=15320249)

File: app/api/custom/orders.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        try:
+            order = Order.query.filter_by(identifier=order_identifier).first()
+        except NoResultFound:
+            return NotFoundError({'source': ''}, 'Order Invoice not found').respond()
+        if current_user.can_download_tickets(order):
+            key = UPLOAD_PATHS['pdf']['order'].format(identifier=order_identifier)
+            file_path = '../generated/invoices/{}/{}/'.format(key, generate_hash(key)) + order_identifier + '.pdf'
+            try:
+                return return_file('invoice', file_path, order_identifier)
+            except FileNotFoundError:
```
Comment (niranjan94) [inline app/api/custom/orders.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4554924)

File: app/api/custom/tickets.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        try:
+            order = Order.query.filter_by(identifier=order_identifier).first()
+        except NoResultFound:
+            return NotFoundError({'source': ''}, 'This ticket is not associated with any order').respond()
+        if current_user.can_download_tickets(order):
+            key = UPLOAD_PATHS['pdf']['tickets_all'].format(identifier=order_identifier)
+            file_path = '../generated/tickets/{}/{}/'.format(key, generate_hash(key)) + order_identifier + '.pdf'
+            try:
+                return return_file('ticket', file_path, order_identifier)
+            except FileNotFoundError:
```
Comment (niranjan94) [inline app/api/custom/tickets.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4554924)

**Comment by prateekj117** [comment]
@kushthedude @iamareebjamal Please review.

**Comment by iamareebjamal** [comment]
Mark the previous PR as depends on

File: app/api/custom/orders.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                create_pdf_tickets_for_holder(order)
+                return return_file('invoice', file_path, order_identifier)
+        else:
+            return ForbiddenError({'source': ''}, 'Unauthorized Access').respond()
+    else:
+        return ForbiddenError({'source': ''}, 'Authentication Required to access Invoice').respond()
```
Comment (iamareebjamal) [inline app/api/custom/orders.py]: This function and the event invoice function should be present in 1 file - invoices.py

**Comment by prateekj117** [comment]
@iamareebjamal Done. Please review.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           

Complexity increasing per file
==============================
- app/api/custom/orders.py  4
- app/api/custom/invoices.py  6
- app/api/custom/tickets.py  5
         

Clones added
============
- app/api/custom/orders.py  2
- app/api/custom/invoices.py  1
- app/api/custom/tickets.py  1
         

Clones removed
==============
+ app/api/auth.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-server/open-event-orga-server/pullRequest?prid=4554924&bid=15320249)

File: app/api/custom/tickets.py (hunk 3 above)
Comment (niranjan94) [inline app/api/custom/tickets.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4554924)

File: app/api/custom/invoices.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        try:
+            order = Order.query.filter_by(identifier=order_identifier).first()
+        except NoResultFound:
+            return NotFoundError({'source': ''}, 'Order Invoice not found').respond()
+        if current_user.can_download_tickets(order):
+            key = UPLOAD_PATHS['pdf']['order'].format(identifier=order_identifier)
+            file_path = '../generated/invoices/{}/{}/'.format(key, generate_hash(key)) + order_identifier + '.pdf'
+            try:
+                return return_file('invoice', file_path, order_identifier)
+            except FileNotFoundError:
```
Comment (niranjan94) [inline app/api/custom/invoices.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4554924)

**Comment by prateekj117** [comment]
Depends on #6610

**Comment by prateekj117** [comment]
@iamareebjamal Please merge this.

**Comment by iamareebjamal** [comment]
There are conflicts

**Comment by prateekj117** [comment]
@iamareebjamal Updating PR in a few minutes.

**Comment by iamareebjamal** [comment]
Rebase with development to have only 1 commit

**Comment by prateekj117** [comment]
@iamareebjamal Rebasing deletes the invoices commit changes.

**Comment by prateekj117** [comment]
@iamareebjamal Solved the conflicts.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           

Complexity increasing per file
==============================
- app/api/custom/invoices.py  6
- app/api/custom/tickets.py  5
         

Complexity decreasing per file
==============================
+ app/api/custom/orders.py  -2
         

Clones added
============
- app/api/custom/invoices.py  1
- app/api/custom/tickets.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-server/open-event-orga-server/pullRequest?prid=4554924&bid=15320249)

File: app/api/custom/invoices.py (hunk 5 above)
Comment (niranjan94) [inline app/api/custom/invoices.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4554924)

File: app/api/custom/tickets.py (hunk 3 above)
Comment (niranjan94) [inline app/api/custom/tickets.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4554924)

File: app/api/custom/tickets.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+from app.api.helpers.storage import UPLOAD_PATHS
+from app.api.helpers.storage import generate_hash
+from app.models.order import Order
+
+ticket_blueprint = Blueprint('ticket_blueprint', __name__, url_prefix='/v1/tickets')
+
+
+@ticket_blueprint.route('/<string:order_identifier>')
+@jwt_required
+def ticket_attendee_authorized(order_identifier):
```
Comment (iamareebjamal) [inline app/api/custom/tickets.py]: This is also order API. Don't know why it is used as tickets API. The tickets in open event are not the same as tickets PDF of an order. 

We'll retain the endpoint for now, but add another endpoint, `/v1/orders/<identifier>/tickets`

**Comment by prateekj117** [comment]
@iamareebjamal Done.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/api/custom/invoices.py  6
- app/api/custom/tickets.py  5
         

Complexity decreasing per file
==============================
+ app/api/custom/orders.py  -1
         

Clones added
============
- app/api/custom/invoices.py  1
- app/api/custom/orders.py  1
- app/api/custom/tickets.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-server/open-event-orga-server/pullRequest?prid=4554924&bid=15320249)

File: app/api/custom/invoices.py (hunk 5 above)
Comment (niranjan94) [inline app/api/custom/invoices.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4554924)

File: app/api/custom/tickets.py (hunk 3 above)
Comment (niranjan94) [inline app/api/custom/tickets.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4554924)

File: app/api/custom/orders.py [hunk 7]
```diff
@@ -19,6 +21,28 @@
 order_blueprint = Blueprint('order_blueprint', __name__, url_prefix='/v1/orders')
 
 
+@order_blueprint.route('/<string:order_identifier>')
```
Comment (iamareebjamal) [inline app/api/custom/orders.py]: You just broke webapp and android app. Read carefully what I wrote

**Comment by prateekj117** [comment]
@iamareebjamal My bad. Sorry for the silly mistake. Please review now.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/api/custom/invoices.py  6
- app/api/custom/tickets.py  5
         

Complexity decreasing per file
==============================
+ app/api/custom/orders.py  -1
         

Clones added
============
- app/api/custom/invoices.py  1
- app/api/custom/orders.py  1
- app/api/custom/tickets.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-server/open-event-orga-server/pullRequest?prid=4554924&bid=15320249)

File: app/api/custom/invoices.py (hunk 5 above)
Comment (niranjan94) [inline app/api/custom/invoices.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4554924)

File: app/api/custom/tickets.py (hunk 3 above)
Comment (niranjan94) [inline app/api/custom/tickets.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4554924)

File: app/api/custom/tickets.py (hunk 6 above)
Comment (iamareebjamal) [reply app/api/custom/tickets.py]: Please don't resolve reviews manually. They're resolved automatically when task is completed

Comment (iamareebjamal) [reply app/api/custom/tickets.py]: Still not moved

Comment (prateekj117) [reply app/api/custom/tickets.py]: @iamareebjamal I didn't delete the older endpoint because it would be in use on frontend and android.

Comment (iamareebjamal) [reply app/api/custom/tickets.py]: You have to move it to orders.py

Without deleting the endpoint. There's no need to duplicate it

Comment (prateekj117) [reply app/api/custom/tickets.py]: @iamareebjamal But what about the other endpoint `/v1/orders/<identifier>/tickets`?

Comment (iamareebjamal) [reply app/api/custom/tickets.py]: That will be added as well

Comment (iamareebjamal) [reply app/api/custom/tickets.py]: Endpoint != Function. So 2 endpoints doesn't mean 2 functions. 1 function can be mapped to many endpoints

**Comment by prateekj117** [comment]
@iamareebjamal Have a look.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/api/custom/invoices.py  6
         

Complexity decreasing per file
==============================
+ app/api/custom/orders.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-server/open-event-orga-server/pullRequest?prid=4554924&bid=15320249)

File: app/api/custom/invoices.py (hunk 5 above)
Comment (niranjan94) [inline app/api/custom/invoices.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4554924)

**Comment by iamareebjamal** [comment]
Verify that it is working correctly

**Comment by prateekj117** [comment]
@iamareebjamal `/v1/orders/<identifier>/tickets` returns all tickets under an order. We need to change the endpoint.

**Comment by iamareebjamal** [comment]
Why does it do that?

**Comment by prateekj117** [comment]
Because this [endpoint](https://open-event-api-dev.herokuapp.com/#tickets-list-tickets-for-an-order-get) already exists. I just need to change the name. Can you suggest me some other name for the endpoint?

**Comment by iamareebjamal** [comment]
tickets-pdf

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/api/custom/invoices.py  6
         

Complexity decreasing per file
==============================
+ app/api/custom/orders.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-server/open-event-orga-server/pullRequest?prid=4554924&bid=15320249)

File: app/api/custom/invoices.py (hunk 5 above)
Comment (niranjan94) [inline app/api/custom/invoices.py]: Codacy Issue found: [undefined name 'FileNotFoundError'](https://app.codacy.com/app/fossasia/open-event-orga-server/pullRequest?prid=4554924)

**Comment by iamareebjamal** [comment]
Hope you've tested it?

**Comment by prateekj117** [comment]
@iamareebjamal Yes.

**Pull request merged**